### PR TITLE
CMSIS-NN: Add Cmake support for unit tests

### PR DESCRIPTION
**Pull request by mansnils**
Adding Cmake support to enable testing on other platforms.
Also Mbed related updates due to updates in Mbed.

### CODE REVIEW

File: CMSIS/NN/Tests/UnitTest/TestCases/test_arm_avgpool_s8/CMakeLists.txt [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+    Unity/TestRunner/unity_test_arm_avgpool_s8_runner.c
+    ${CMSIS_PATH}/CMSIS/NN/Source/PoolingFunctions/arm_avgpool_s8.c)
+
+target_include_directories(test_arm_avgpool_s8 PRIVATE ${CMSIS_PATH}/CMSIS/NN/Include)
+target_include_directories(test_arm_avgpool_s8 PRIVATE ${CMSIS_PATH}/CMSIS/DSP/Include)
+target_include_directories(test_arm_avgpool_s8 PRIVATE ${CMSIS_PATH}/CMSIS/Core/Include)
+
+target_link_libraries(test_arm_avgpool_s8 LINK_PUBLIC unity)
+
+add_dependencies(test_arm_avgpool_s8 unity)
```
Comment (christophe0606) [inline CMSIS/NN/Tests/UnitTest/TestCases/test_arm_avgpool_s8/CMakeLists.txt:32]: Is it needed with the target_link_libraries before ?

File: CMSIS/NN/Tests/UnitTest/TestCases/test_arm_convolve_1x1_s8_fast/CMakeLists.txt [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+    ${CMSIS_PATH}/CMSIS/NN/Source/ConvolutionFunctions/arm_nn_mat_mult_kernel_s8_s16.c)
+
+target_include_directories(test_arm_convolve_1x1_s8_fast PRIVATE ${CMSIS_PATH}/CMSIS/NN/Include)
+target_include_directories(test_arm_convolve_1x1_s8_fast PRIVATE ${CMSIS_PATH}/CMSIS/DSP/Include)
+target_include_directories(test_arm_convolve_1x1_s8_fast PRIVATE ${CMSIS_PATH}/CMSIS/Core/Include)
+
+target_link_libraries(test_arm_convolve_1x1_s8_fast LINK_PUBLIC arm_nn_mat_mult_core)
+target_link_libraries(test_arm_convolve_1x1_s8_fast LINK_PUBLIC unity)
+
+add_dependencies(test_arm_convolve_1x1_s8_fast arm_nn_mat_mult_core unity)
```
Comment (christophe0606) [inline CMSIS/NN/Tests/UnitTest/TestCases/test_arm_convolve_1x1_s8_fast/CMakeLists.txt:40]: Is it needed with the target_link_libraries ?

**Review by christophe0606 — APPROVED** [review]

File: CMSIS/NN/Tests/UnitTest/TestCases/test_arm_avgpool_s8/CMakeLists.txt (hunk 1 above)
Comment (mansnils) [reply CMSIS/NN/Tests/UnitTest/TestCases/test_arm_avgpool_s8/CMakeLists.txt:32]: No it seems not. Will remove it.

File: CMSIS/NN/Tests/UnitTest/TestCases/test_arm_convolve_1x1_s8_fast/CMakeLists.txt (hunk 2 above)
Comment (mansnils) [reply CMSIS/NN/Tests/UnitTest/TestCases/test_arm_convolve_1x1_s8_fast/CMakeLists.txt:40]: Removing it

**Pull request merged**